cloudflare/resource_cloudflare_fallback_domain_test.go

```diff
@@ -0,0 +1,64 @@
+package cloudflare
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccCloudflareFallbackDomain(t *testing.T) {
+	// Temporarily unset CLOUDFLARE_API_TOKEN if it is set as the Access
+	// service does not yet support the API tokens and it results in
+	// misleading state error messages.
+	if os.Getenv("CLOUDFLARE_API_TOKEN") != "" {
+		defer func(apiToken string) {
+			os.Setenv("CLOUDFLARE_API_TOKEN", apiToken)
+		}(os.Getenv("CLOUDFLARE_API_TOKEN"))
+		os.Setenv("CLOUDFLARE_API_TOKEN", "")
+	}
+
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("cloudflare_fallback_domain.%s", rnd)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccessAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareFallbackDomain(rnd, accountID, "example domain", "example.com", []string{"2.2.2.2"}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "account_id", accountID),
+					resource.TestCheckResourceAttr(name, "domains.0.description", "example domain"),
+					resource.TestCheckResourceAttr(name, "domains.0.suffix", "example.com"),
+					resource.TestCheckResourceAttr(name, "domains.0.dns_server", "['2.2.2.2']"),
```

Comment:
i think you want the first entry in the slice here

```suggestion
					resource.TestCheckResourceAttr(name, "domains.0.dns_server.0", "2.2.2.2"),
```